### PR TITLE
CMake changes to clean up conda-forge dependencies

### PR DESCRIPTION
**Pull request by bennibbelink**
See https://github.com/conda-forge/cyclus-feedstock/pull/80 and https://github.com/conda-forge/cyclus-feedstock/pull/81 for more dialogue about these changes.  Notably:
- remove `pandas` as a build-time dependency
- update how we link against `libxml++` libs
- update how we find `blas` libs
- use `-` consistently in `install.py` flags instead of `_` (might necessitate some documentation updates in the website)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -331,7 +330,6 @@ IF(NOT CYCLUS_DOC_ONLY)
         include(UseCython)
         find_package (Python3 REQUIRED COMPONENTS NumPy)
         find_package(Jinja2 REQUIRED)
```
Comment (gonuke) [inline CMakeLists.txt:332]: Is jinja also run-time only?

Comment (bennibbelink) [reply CMakeLists.txt:332]: No, `jinja2` is used by `gentypesystem.py` during the build process

File: CMakeLists.txt [hunk 2]
```diff
@@ -207,7 +208,6 @@ IF(NOT CYCLUS_DOC_ONLY)
       link_directories(${HDF5_LIBRARY_DIRS})
     endif()
     set(LIBS ${LIBS} ${HDF5_C_LIBRARIES} ${HDF5_C_HL_LIBRARIES})
-    MESSAGE("--    HDF5 Root: ${HDF5_ROOT}")
```
Comment (gonuke) [inline CMakeLists.txt:210]: Do we want to delete this message? (and the BOOST one?)

**Review by gonuke — CHANGES_REQUESTED** [review]
One question about messages

File: CMakeLists.txt (hunk 2 above)
Comment (bennibbelink) [reply CMakeLists.txt]: I added them back in.

**Review by gonuke — APPROVED** [review]
Thanks @bennibbelink - LGTM

**Pull request merged**